soda/scientific/soda/scientific/distribution/comparison.py

```diff
@@ -99,27 +110,22 @@ def _parse_reference_cfg(self, ref_file_path: FilePath) -> RefDataCfg:
         return RefDataCfg.parse_obj(ref_data_cfg)
 
 
-class DistributionAlgorihm:
+class DistributionAlgorithm(abc.ABC):
     def __init__(self, cfg: RefDataCfg, test_data: pd.Series, seed: int = 61) -> None:
-        self.cfg = cfg
         self.test_data = test_data
-        self.rng = np.random.default_rng(seed)
-        self.ref_data = self.generate_ref_data()
-
-    @abc.abstractmethod
-    def evaluate(self, test_data: pd.Series) -> Tuple[float, float]:
-        ...
+        self.ref_data = generate_ref_data(cfg, len(test_data), np.random.default_rng(seed))
 
     @abc.abstractmethod
-    def generate_ref_data(self) -> pd.Series:
+    def evaluate(self) -> dict:
         ...
 
 
-class ChiSqAlgorithm(DistributionAlgorihm):
+class ChiSqAlgorithm(DistributionAlgorithm):
     def __init__(self, cfg: RefDataCfg, test_data: pd.Series, seed: int = 61) -> None:
+        cfg.dtype = "categorical"
```

Comment:
I don't get why we go and modify the `dtype` of the cfg in here? Isn't it supposed to be something that the user has to provide?